File: java/socketd/src/main/java/org/noear/socketd/transport/stream/RequestStream.java
package org.noear.socketd.transport.stream;

import org.noear.socketd.transport.core.Reply;
import org.noear.socketd.utils.IoConsumer;

/**
 * 请求流
 *
 * @author noear
 * @since 2.3
 */
public interface RequestStream extends Stream<RequestStream> {
    /**
     * 异步等待获取答复
     */
    Reply await();

    /**
     * 答复发生时
     */
    RequestStream thenReply(IoConsumer<Reply> onReply);
}

File: swift/socketd/socketd/src/socketd/transport/stream/RequestStream.swift
//
//  RequestStream.swift
//  socketd
//
//  Created by noear on 2024/1/12.
//

import Foundation


/**
 * 请求流
 *
 * @author noear
 * @since 2.3
 */
protocol RequestStream : Stream{
    
    /**
     * 异步等待获取答复
     */
    func await() -> Reply?;
    
    /**
     * 答复发生时
     */
    func  thenReply(_ onReply:IoConsumer<Reply>) -> Self;
}
